spec/helper/revenuecat_internal_helper_spec.rb

```diff
@@ -0,0 +1,306 @@
+      end.to raise_exception(StandardError)
+    end
+
+    it 'fails if user cancels on confirmation to open editor' do
+      expect(File).not_to receive(:write)
+      allow(FastlaneCore::UI).to receive(:confirm).with('Open CHANGELOG.latest.md in \'vim\'? (No will quit this process)').and_return(false)
+      expect do
+        Fastlane::Helper::RevenuecatInternalHelper.edit_changelog(prepopulated_changelog, changelog_latest_path, editor)
+      end.to raise_exception(StandardError)
+    end
+
+    it 'asks for confirmation if prepopulated changelog remains the same after editor opening' do
+      allow(File).to receive(:read).with(changelog_latest_path).and_return(prepopulated_changelog)
+      expect(FastlaneCore::UI).to receive(:confirm)
+        .with('You may have opened the changelog in a visual editor. Enter \'y\' when changes are saved or \'n\' to cancel').and_return(true).once
+      Fastlane::Helper::RevenuecatInternalHelper.edit_changelog(prepopulated_changelog, changelog_latest_path, editor)
+    end
+
+    it 'fails if confirmation if prepopulated changelog remains the same after editor opening' do
+      allow(File).to receive(:read).with(changelog_latest_path).and_return(prepopulated_changelog)
+      expect(FastlaneCore::UI).to receive(:confirm)
+        .with('You may have opened the changelog in a visual editor. Enter \'y\' when changes are saved or \'n\' to cancel').and_return(false).once
+      expect do
+        Fastlane::Helper::RevenuecatInternalHelper.edit_changelog(prepopulated_changelog, changelog_latest_path, editor)
+      end.to raise_exception(StandardError)
+    end
+  end
+
+  describe '.attach_changelog_to_master' do
+    require 'fileutils'
+
+    let(:version_number) { '1.12.0' }
+    let(:tmp_test_files_path) { './tmp_test_files' }
+    let(:changelog_latest_path) { './tmp_test_files/CHANGELOG.latest.md' }
+    let(:changelog_path) { './tmp_test_files/CHANGELOG.md' }
+
+    before(:each) do
+      Dir.mkdir(tmp_test_files_path)
+      File.write(changelog_latest_path, 'changelog latest contents')
+      File.write(changelog_path, "## 1.11.0\nchangelog contents")
+    end
+
+    after(:each) do
+      FileUtils.rm_rf(tmp_test_files_path)
+    end
+
+    it 'prepends changelog latest file contents to changelog file' do
+      Fastlane::Helper::RevenuecatInternalHelper.attach_changelog_to_master(version_number, changelog_latest_path, changelog_path)
+      changelog_contents = File.read(changelog_path)
+      expect(changelog_contents).to eq("## 1.12.0\nchangelog latest contents\n## 1.11.0\nchangelog contents")
+    end
+  end
+
+  describe '.create_new_release_branch' do
+    it 'creates new release branch with version number' do
+      expect(Fastlane::Actions).to receive(:sh).with("git checkout -b 'release/1.12.0'")
+      Fastlane::Helper::RevenuecatInternalHelper.create_new_release_branch('1.12.0')
+    end
+  end
```

Comment:
I think you could do another test that checks if the branch already exists on remote. If it does, we should at least halt deployment.

Reply:
Right now I was just copying the existing logic from iOS (which has the most advanced automation IMO) where it wouldn't fail in that case but I can add that logic since it shouldn't be difficult. Will do so in another PR though if that's ok so it can be reviewed better. I will do that at the beginning of the`BumpVersionUpdateChangelogCreatePrAction` since it's where we create the release branch.

Reply:
Added this validation here: https://github.com/RevenueCat/fastlane-plugin-revenuecat_internal/pull/5

Reply:
Love it, thank you!